src/cdk/overlay/position/viewport-margin.ts

```diff
@@ -0,0 +1 @@
+export type ViewportMargin = number | {top?: number; bottom?: number; start?: number; end?: number};
```

Comment:
It looks like this file is missing a license header. You can also put the type in one of the existing files.